src/cms.js

```diff
@@ -19,13 +19,13 @@ class CMS {
       return callback(new Error('Data is required'))
     }
 
-    self.keystore._getPrivateKey(name, (err, key) => {
+    self.keychain._getPrivateKey(name, (err, key) => {
       if (err) {
         return callback(err)
       }
 
       try {
-        const privateKey = forge.pki.decryptRsaPrivateKey(key, self.keystore._())
+        const privateKey = forge.pki.decryptRsaPrivateKey(key, self.keychain._())
```

Comment:
Please avoid node-forge. We used it before and it is a _HUGE_ dependency. Also last time I checked it was using UMD instead of CJS or ES6 modules which makes it a PITA to import.

Reply:
See https://github.com/libp2p/js-libp2p-keychain/issues/7

Reply:
Still on #7, the crypto primitives should come from the libp2p-crypto. Let's then figure out if we can just extract these functions out of node-forge or another package on npm.

Reply:
Enhancing libp2p-crypto will take time.  It needs to 
- create/read PKCS 8 (https://github.com/libp2p/js-libp2p-crypto/issues/114)
- create/read PKCS 7 CMS
- support PBKDF2
- support various hash and HMAC algorithms
- support OID
- support ASN.1
- ...

Crypto is a big issue.

My game plan is to initially use another package (node.forge) so that `ipfs key` and `ipfs crypto` can be implemented.  Then we can gradually enhance libp2p2-crypto.

Reply:
> My game plan is to initially use another package (node.forge) so that ipfs key and ipfs crypto can be implemented. Then we can gradually enhance libp2p2-crypto.

Yes, we can use another package. Note, libp2p-crypto is just a bundle of other modules, we do not implement any crypto functions ourselves. However, for questions of consistency, we want libp2p crypto to expose the crypto primitives.

Can you add these funcs you need there?